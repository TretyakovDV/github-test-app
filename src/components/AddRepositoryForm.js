import React from 'react';
import PropTypes from 'prop-types';

const AddRepositoryForm = ({ onSubmit }) => {
  const [name, setName] = React.useState('facebook/react');

  const handleChange = ({ target: { value } }) => setName(value);
  const handleClick = () => onSubmit(name);

  return (
    <div>
      <input
        type="text"
        placeholder="facebook/react"
        value={name}
        onChange={handleChange}
      />
      <button
        type="button"
        onClick={handleClick}
      >
        Add
      </button>
    </div>
  );
};

AddRepositoryForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

export default AddRepositoryForm;
