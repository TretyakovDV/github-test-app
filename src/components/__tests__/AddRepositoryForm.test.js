import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import AddRepositoryForm from '../AddRepositoryForm';

describe('AddRepositoryForm', () => {
  it('matches snapshot', () => {
    const onSubmit = jest.fn();

    const { asFragment } = render(<AddRepositoryForm onSubmit={onSubmit} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('onSubmit()', () => {
    const onSubmit = jest.fn();

    const { getByPlaceholderText, getByText } = render(<AddRepositoryForm onSubmit={onSubmit} />);
    fireEvent.change(getByPlaceholderText('facebook/react'), 'facebook/react');
    fireEvent.click(getByText('Add'));

    expect(onSubmit).toHaveBeenCalledTimes(1);
    expect(onSubmit).toHaveBeenCalledWith('facebook/react');
  });
});
