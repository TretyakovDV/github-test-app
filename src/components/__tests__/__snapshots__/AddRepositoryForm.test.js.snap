// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRepositoryForm matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <input
      placeholder="facebook/react"
      type="text"
      value="facebook/react"
    />
    <button
      type="button"
    >
      Add
    </button>
  </div>
</DocumentFragment>
`;
